apis/k8ssandra/v1alpha1/k8ssandracluster_webhook.go

```diff
@@ -85,14 +89,107 @@ func (r *K8ssandraCluster) validateK8ssandraCluster() error {
 			}
 		}
 	}
+	if err := TelemetrySpecsAreValid(r, clientCache); err != nil {
+		return err
+	}
+	return nil
+}
+
+// clientGetter is an interface here purely to enable DI. It is anything that implements `GetRemoteClient()` mechanism to get remote clients by name.
+type clientGetter interface {
```

Comment:
This is unnecessary, we already have ``clientCache.GetRemoteClient(dc.K8sContext)`` available here and it's available in the envtest also. There isn't need for another interface that hides the implementation.

Reply:
This required to sub in a mock implementation. If this is confusing please feel free to discuss with me offline.